### ISSUE TITLE
[Feature Request] Option to set fans to 100% on application exit

### ISSUE THREAD

**Issue by Squire136** [Rem0o/FanControl.Releases#1268]
SpeedFan had an option I really liked. You could check a box and have all fans ramp to 100% on application exit.

This was useful if the application exited unintentionally or I needed to shut down the application while troubleshooting other issues.

It would mean I don't have to worry about my CPU or GPU overheating if I'm forced to close the application.

Even if it's 80% or 90% on all fans on application exit, but it would just be helpful and hopefully isn't a lot of work to add a feature like that.

Thanks.

**Comment by Rem0o** [comment]
> This was useful if the application exited unintentionally

If it exits unintentionally, it won't be able to apply that logic, since it got most likely process killed.

> It would mean I don't have to worry about my CPU or GPU overheating if I'm forced to close the application. 

When Fan Control exits, it resets to stock behavior, which is set by default to prevent such thing.  It's not like all fans turned down to 0.  I'm not sure the scenario you are implying actually exists.

**Comment by bradsm87** [comment]
> > This was useful if the application exited unintentionally
> 
> If it exits unintentionally, it won't be able to apply that logic, since it got most likely process killed.
> 
> > It would mean I don't have to worry about my CPU or GPU overheating if I'm forced to close the application.
> 
> When Fan Control exits, it resets to stock behavior, which is set by default to prevent such thing. It's not like all fans turned down to 0. I'm not sure the scenario you are implying actually exists.

Some motherboards may work like that but I know mine doesn't. If I exit FanControl at idle, the fans stay that speed until the next restart or until FanControl opens again. If I load up my CPU, it hits its thermal limit straight away and the fans do not speed up.

**Comment by Rem0o** [comment]
@bradsm87 I would recommend playing a bit with your fan bios settings and try to get a better default behaviour. You can most likely have stock curves running at least on your cpu fan.

**Comment by bradsm87** [comment]
> @bradsm87 I would recommend playing a bit with your fan bios settings and try to get a better default behaviour. You can most likely have stock curves running at least on your cpu fan.

Yes that's my workaround for now. I use the BIOS fan control for the main CPU fan so even if FanControl is closed for whatever reason, the man CPU fan can still ramp up under load. The two other fans are controlled by FanControl and remain at idle (not the BIOS setting for those fans) if FanControl is closed. These fans work according to the BIOS setting before FanControl is started but the BIOS does not take over control after FanControl is closed.

**Comment by Rem0o** [comment]
Can’t fan control override your cpu fan if the bios has a default curve applied to it? Like in mu personal case, cpu is set to auto in bios, and FanControl can take over. When I exit FC, the bios behaviour is restored.  By the way, NEVER force close the app with task manager. Always use the exit option.

**Comment by bradsm87** [comment]
> Can’t fan control override your cpu fan if the bios has a default curve applied to it? Like in mu personal case, cpu is set to auto in bios, and FanControl can take over. When I exit FC, the bios behaviour is restored. By the way, NEVER force close the app with task manager. Always use the exit option.

Yes FanControl does override the BIOS settings when FanControl launches. No it does not return to BIOS settings when FanControl is closed using the exit option. The fans stay at the last speed used by FanControl. They do not revert to BIOS settings when exiting FanControl.

**Comment by Rem0o** [comment]
What exactly do you got set for your CPU fan in your bios for example?

**Comment by bradsm87** [comment]
> What exactly do you got set for your CPU fan in your bios for example?

I could set the fan to fixed 80% in BIOS for example. It will be at 80% during boot-up. When FanControl launches, it sets it to 36% at idle. If FanControl is exited at idle, fan will stay at 36% (last FanControl state) indefinitely until the next restart or until FanControl is opened again.

**Comment by bradsm87** [comment]
> Instead of a fixed 80, do you have any “auto” option that FanControl can still override?

80 was just an example. One of the fans for example is set in the BIOS to start at 60% at 46C CPU temp and linear to 90% at 85C CPU temp. It does this as expected before FanControl launches. FanControl takes over and runs it quieter than that, 50% at idle for example. If I exit FanControl at idle, that fan will stay at 50%, even if the CPU is 91C and it will only return to the BIOS fan control setting after a shut down or restart.

What I’m saying is not all motherboards resume BIOS fan control settings after software fan control stops. Some motherboards (including mine) do not revert to motherboard fan control settings until next boot.

**Comment by Rem0o** [comment]
@bradsm87 I mentionned it specifically because the software on exit does call a "revert the header to default" function. 
It does the exact same thing as if you disabled a specific control when the app is running.  

Indeed it may seem like that doesn't work on all motherboards.   

Just curious though, if you disable a control, does it stays stuck to whatever % is was ?

**Comment by bradsm87** [comment]
> @bradsm87 I mentionned it specifically because the software on exit does call a "revert the header to default" function. It does the exact same thing as if you disabled a specific control when the app is running.
> 
> Indeed it may seem like that doesn't work on all motherboards.
> 
> Just curious though, if you disable a control, does it stays stuck to whatever % is was ?

I'll try that. My computer is packed away for the next week during some renovations.

**Comment by TheVillageGuy** [comment]
I agree this is a feature that was on speedfan and am now missing. It had 50% on exit which worked like a charm
